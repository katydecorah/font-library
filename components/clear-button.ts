class ClearButton extends HTMLElement {
  constructor() {
    super();
  }
  connectedCallback() {
    // add aria-role
    this.setAttribute("role", "button");
    this.addEventListener("click", () => {
      this.dispatchEvent(
        new CustomEvent("clear-filter", {
          bubbles: true,
          composed: true,
          detail: {
            filter: this.getAttribute("value"),
          },
        })
      );
    });
  }
}

customElements.define("clear-button", ClearButton);
