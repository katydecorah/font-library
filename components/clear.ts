class ClearButton extends HTMLElement {
  constructor() {
    super();
  }
  connectedCallback() {
    this.addEventListener("click", () => {
      this.dispatchEvent(
        new CustomEvent("clear-filter", {
          bubbles: true,
          composed: true,
        })
      );
    });
  }
}

customElements.define("clear-button", ClearButton);
