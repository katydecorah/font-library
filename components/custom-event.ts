export default function customEvent(
  eventName: string,
  detail?: {
    value?: string | boolean | number;
    id?: string;
  }
): CustomEvent {
  return new CustomEvent(eventName, {
    bubbles: true,
    composed: true,
    detail,
  });
}
