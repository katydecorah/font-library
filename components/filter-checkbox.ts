import customEvent from "./custom-event";

class FilterCheckbox extends HTMLInputElement {
  constructor() {
    super();
    this.addEventListener("change", this.onChange);
    this.handleInitialValue();

    // Listen for events to clear filter
    window.addEventListener("remove-checkbox", () => {
      this.checked = false;
      this.onChange();
    });
  }

  onChange() {
    this.dispatchEvent(
      customEvent("handle-filter", {
        value: this.checked,
        id: this.id,
      })
    );
    this.setUrlParam();
  }

  setUrlParam() {
    const urlParameters = new URLSearchParams(window.location.search);
    // only set variable if it's true
    if (this.checked === false) {
      urlParameters.delete("variable");
    } else {
      urlParameters.set("variable", "true");
    }
    window.history.replaceState(
      {},
      "",
      `${window.location.pathname}?${urlParameters.toString()}`
    );
  }

  handleInitialValue() {
    const urlParameters = new URLSearchParams(window.location.search);
    const initialValue = urlParameters.get("variable");
    if (initialValue === "true") {
      this.checked = true;

      // Wait for main-app to load before dispatching event
      window.addEventListener("main-app-loaded", () => {
        this.onChange();
      });
    }
  }
}

customElements.define("filter-checkbox", FilterCheckbox, { extends: "input" });
