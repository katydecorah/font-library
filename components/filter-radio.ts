class FilterRadio extends HTMLInputElement {
  constructor() {
    super();
    this.onchange = this.onChange;

    // Get init value from URL param
    const urlParams = new URLSearchParams(window.location.search);
    const tag = urlParams.get("tag");
    if (tag === this.value) {
      this.checked = true;
    }
  }

  onChange() {
    this.dispatchEvent(
      new CustomEvent("tag-button-selected", {
        bubbles: true,
        composed: true,
        detail: {
          value: this.value,
        },
      })
    );

    this.setUrlParam();
  }

  setUrlParam() {
    const urlParams = new URLSearchParams(window.location.search);
    urlParams.set("tag", this.value.toString());
    window.history.replaceState(
      {},
      "",
      `${window.location.pathname}?${urlParams.toString()}`
    );
  }
}

customElements.define("filter-radio", FilterRadio, { extends: "input" });
