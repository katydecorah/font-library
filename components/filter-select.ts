class FilterSelect extends HTMLSelectElement {
  constructor() {
    super();
    this.onchange = this.onChange;

    // Get initial value from URL
    const urlParams = new URLSearchParams(window.location.search);
    const initValue = urlParams.get(this.dataset.param);
    if (initValue) {
      if (!this.options.namedItem(initValue)) return;
      this.value = initValue;

      // Wait for main-app to load before dispatching event
      window.addEventListener("main-app-loaded", () => {
        this.onChange();
      });
    }

    // Listen for events to clear filter
    window.addEventListener("remove-select", (e: CustomEvent) => {
      if (e.detail.filter === this.id) {
        this.value = "";
        this.onChange();
      }
    });

    // Listen for changes by other tag elements (radio, button)
    window.addEventListener("tag-button-selected", (e: CustomEvent) => {
      if (this.id === "selectedTag" && this.value !== e.detail.value) {
        this.value = e.detail.value;
      }
    });
  }

  onChange() {
    const eventName =
      this.id === "selectedTag" ? "tag-button-selected" : "handle-filter";
    this.dispatchEvent(
      new CustomEvent(eventName, {
        bubbles: true,
        composed: true,
        detail: { id: this.id, value: this.value },
      })
    );

    this.setUrlParam();
  }

  setUrlParam() {
    const param = this.dataset.param;
    const urlParams = new URLSearchParams(window.location.search);
    if (!this.value) {
      urlParams.delete(param);
    } else {
      urlParams.set(param, this.value.toString());
    }
    window.history.replaceState(
      {},
      "",
      `${window.location.pathname}?${urlParams.toString()}`
    );
  }
}

customElements.define("filter-select", FilterSelect, { extends: "select" });
