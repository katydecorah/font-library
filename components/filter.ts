import { GeneratedData } from "./main-app";

export default function filter(
  state: {
    selectedSearch: string;
    selectedTag: string;
    selectedCategory: string;
    selectedSubset: string;
    selectedVariant: string;
    selectedVariable: boolean;
    sortBy: string;
    currentPage: number;
    pageSize: number;
  },
  generatedData: GeneratedData
): [number, GeneratedData] {
  const {
    selectedSearch,
    selectedTag,
    selectedCategory,
    selectedSubset,
    selectedVariant,
    selectedVariable,
    sortBy,
    currentPage,
    pageSize,
  } = state;
  let filteredData = generatedData;

  if (selectedSearch) {
    filteredData = filteredData.filter((row) =>
      row.family.toLowerCase().includes(selectedSearch.toLowerCase())
    );
  }

  if (selectedTag && selectedTag !== "need tags") {
    filteredData = filteredData.filter((row) => row.tags.includes(selectedTag));
  }
  if (selectedTag === "need tags") {
    filteredData = filteredData.filter((row) => row.tags.length === 0);
  }
  if (selectedCategory) {
    filteredData = filteredData.filter(
      (row) => row.category === selectedCategory
    );
  }
  if (selectedSubset) {
    filteredData = filteredData.filter((row) =>
      row.subsets.includes(selectedSubset)
    );
  }
  if (selectedVariant) {
    filteredData = filteredData.filter((row) =>
      row.variants.includes(selectedVariant)
    );
  }

  if (selectedVariable === true) {
    filteredData = filteredData.filter((row) => row.variable);
  }

  if (sortBy === "date") {
    filteredData = filteredData.sort(
      (a, b) =>
        new Date(b.lastModified).getTime() - new Date(a.lastModified).getTime()
    );
  }

  if (sortBy === "family") {
    filteredData = filteredData.sort((a, b) =>
      a.family.localeCompare(b.family)
    );
  }

  const paginated = filteredData.filter((row, index) => {
    const start = (currentPage - 1) * pageSize;
    const end = currentPage * pageSize;
    if (index >= start && index < end) return true;
  });

  return [filteredData.length, paginated];
}
