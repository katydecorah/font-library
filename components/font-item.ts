import sampleSubsets from "../data/samples.json";
import rtlSubsets from "../data/rtl.json";
import swaps from "../data/swaps.json";

export type SampleSubsets = typeof sampleSubsets;
export type RtlSubsets = typeof rtlSubsets;
export type Swaps = typeof swaps;

class FontItem extends HTMLElement {
  family: string;
  category: string;
  variants: string[];
  subsets: string[];
  lineNumber: string;
  tags: string[];
  selectedSubset: string;
  subset: string;
  selectedVariant: string;
  id: string;
  slug: string;

  constructor() {
    super();
  }

  connectedCallback() {
    const fontString = this.getAttribute("font");
    const selectedTag = this.getAttribute("selectedTag");
    const selectedSubset = this.getAttribute("selectedSubset");
    this.selectedSubset = selectedSubset;
    const selectedVariant = this.getAttribute("selectedVariant");
    this.selectedVariant = selectedVariant;
    this.subset = this.selectedSubset;

    const font = JSON.parse(fontString);
    const { family, category, variants, subsets, lineNumber, tags, variable } =
      font;
    this.id = family.toLowerCase().replace(/ /g, "-");
    this.slug = family.replace(/ /g, "+");
    const previewName = this.subsetFamily(font);

    this.addFontToHead({ previewName, ...font });

    const familyStyle = this.familyStyle({ previewName, ...font });

    this.innerHTML = `<div id="family-${this.id}" class="family">
    <div class="family-link">
      <div id="family-name" class="family-title ${
        this.id
      }" style="${familyStyle}">
        ${previewName}
      </div>
      <div class="family-meta-container">
     <span class="family-title-small">${
       previewName == family ? "" : family
     }</span>
      <div class="family-meta">
        <span>${category}</span>
        &bull;
        <span aria-label="${variants.join(", ")}"
          >${variants.length} variants${variable ? " (variable)" : ""}</span
        >
        &bull;
        <span aria-label="${subsets.join(", ")}">${
      subsets.length
    } subsets</span>
      </div>
      </div>
    </div>
    <div class="family-tags">
      <div class="family-tags-container">${tags
        .map(
          (tag: string) =>
            `<button is="tag-button" selectedTag="${selectedTag}" value="${tag}">${tag}</button>`
        )
        .join("")}</div>
      <div class="family-meta-links">
        <a
          href="https://github.com/katydecorah/font-library/blob/gh-pages/families.json#L${lineNumber}"
          target="_blank"
          aria-label="Edit tags for ${family}"
          >Edit tags</a
        >
        <a
          href="https://fonts.google.com/specimen/${this.slug}"
          target="_blank"
          aria-label="Visit ${family} on Google Fonts"
          >Google Fonts &rarr;</a
        >
      </div>
    </div>
  </div>`;
  }

  addFontToHead({
    previewName,
    family,
  }: {
    previewName: string;
    family: string;
  }): void {
    const googleFont = document.createElement("link");
    googleFont.href = `https://fonts.googleapis.com/css2?family=${this.fontCall(
      { previewName }
    )}`;
    googleFont.rel = "stylesheet";
    googleFont.setAttribute("data-family", family);
    document.head.appendChild(googleFont);
  }

  fontCall({ previewName }: { previewName: string }): string {
    let font = this.slug;

    const hasItalic = this.selectedVariant.includes("italic");
    const variantNumber = this.selectedVariant.match(/\d+/g); // get number from selectedVariant

    if (this.selectedVariant && this.selectedVariant !== "regular") {
      const variants = [];
      if (hasItalic) {
        variants.push("ital");
      }
      if (variantNumber && variantNumber[0]) {
        variants.push(`wght@${hasItalic ? "1," : ""}${variantNumber[0]}`);
      }
      font += `:${variants.join(",")}`;
    }

    font += `&text=${encodeURIComponent(previewName)}`;

    font += `&display=swap`;

    return font;
  }

  familyStyle({
    previewName,
    family,
  }: {
    previewName: string;
    subsets: string[];
    family: string;
  }): string {
    let style = `font-family: '${family}';`;
    if (rtlSubsets.includes(this.subset) && family !== previewName) {
      style += "direction: rtl;";
    }
    if (this.selectedVariant.includes("italic")) {
      style += "font-style: italic;";
    }
    return style;
  }

  subsetFamily({
    family,
    subsets,
  }: {
    family: string;
    subsets: string[];
  }): string {
    if (this.selectedSubset && this.selectedSubset in sampleSubsets) {
      return sampleSubsets[this.selectedSubset as keyof SampleSubsets];
    }

    if (family in swaps) {
      return swaps[family as keyof Swaps];
    }

    if (
      (!subsets.includes("latin") || family.startsWith("Noto")) &&
      sampleSubsets[subsets[0] as keyof SampleSubsets]
    ) {
      this.subset = subsets[0];
      return sampleSubsets[subsets[0] as keyof SampleSubsets];
    }
    return family;
  }
}

customElements.define("font-item", FontItem);
