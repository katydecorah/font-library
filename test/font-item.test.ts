import "../components/font-item";

describe("FontItem", () => {
  test("renders correctly", () => {
    document.body.innerHTML = `<font-item selectedVariant="" selectedSubset="" font='{"family":"Alef","variants":["regular","700"],"subsets":["hebrew","latin"],"category":"sans-serif","tags":["geometric"],"lineNumber":23}'></font-item>`;
    const fontItem = document.querySelector("font-item");
    expect(fontItem).toMatchSnapshot();
  });

  test("no latin subset", () => {
    document.body.innerHTML = `<font-item selectedVariant="" selectedSubset="" font='{"family":"GFS Neohellenic","variants":["regular","italic","700","700italic"],"subsets":["greek"],"category":"sans-serif","tags":["round","monolinear","1930s"],"lineNumber":439}'></font-item>`;
    const fontItem = document.querySelector("font-item");
    expect(fontItem).toMatchSnapshot();
  });

  test("rtl subset", () => {
    document.body.innerHTML = `<font-item selectedVariant="" selectedSubset="" font='{"family":"Almarai","variants":["300","regular","700","800"],"subsets":["arabic"],"category":"sans-serif","tags":["arabic"],"lineNumber":41}'></font-item>`;
    const fontItem = document.querySelector("font-item");
    expect(fontItem).toMatchSnapshot();
  });

  test("selected variant, 900italic", () => {
    document.body.innerHTML = `<font-item selectedVariant="900italic" selectedSubset="" font='{"family":"Anybody","variants":["100","200","300","regular","500","600","700","800","900","100italic","200italic","300italic","italic","500italic","600italic","700italic","800italic","900italic"],"subsets":["latin","latin-ext","vietnamese"],"category":"display","tags":["eurostile"],"lineNumber":78}'></font-item>`;
    const fontItem = document.querySelector("font-item");
    expect(fontItem).toMatchSnapshot();
  });

  test("selected variant, 900", () => {
    document.body.innerHTML = `<font-item selectedVariant="900" selectedSubset="" font='{"family":"Anybody","variants":["100","200","300","regular","500","600","700","800","900","100italic","200italic","300italic","italic","500italic","600italic","700italic","800italic","900italic"],"subsets":["latin","latin-ext","vietnamese"],"category":"display","tags":["eurostile"],"lineNumber":78}'></font-item>`;
    const fontItem = document.querySelector("font-item");
    expect(fontItem).toMatchSnapshot();
  });

  test("selected subset", () => {
    document.body.innerHTML = `<font-item selectedVariant="" selectedSubset="hebrew" font='{"family":"Alef","variants":["regular","700"],"subsets":["hebrew","latin"],"category":"sans-serif","tags":["geometric"],"lineNumber":23}'></font-item>`;
    const fontItem = document.querySelector("font-item");
    expect(fontItem).toMatchSnapshot();
  });

  test("has family name that exists in swaps.json", () => {
    document.body.innerHTML = `<font-item selectedVariant="" selectedSubset="" font='{"family":"Material Symbols Sharp","variants":["100","200","300","regular","500","600","700"],"subsets":["latin"],"category":"monospace","tags":[],"lineNumber":752}'></font-item>`;
    const fontItem = document.querySelector("font-item");
    expect(fontItem).toMatchSnapshot();
  });
});
