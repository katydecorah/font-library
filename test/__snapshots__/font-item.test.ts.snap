// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FontItem has family name that exists in swaps.json 1`] = `
<font-item
  font="{"family":"Material Symbols Sharp","variants":["100","200","300","regular","500","600","700"],"subsets":["latin"],"category":"monospace","tags":[],"lineNumber":752}"
  id="material-symbols-sharp"
  selectedsubset=""
  selectedvariant=""
>
  <div
    class="family"
    id="family-material-symbols-sharp"
  >
    
    
    <div
      class="family-link"
    >
      
      
      <div
        class="family-title material-symbols-sharp"
        id="family-name"
        style="font-family: 'Material Symbols Sharp';"
      >
        
        favorite add delete settings search
      
      </div>
      
      
      <div
        class="family-meta-container"
      >
        
     
        <span
          class="family-title-small"
        >
          Material Symbols Sharp
        </span>
        
      
        <div
          class="family-meta"
        >
          
        
          <span>
            monospace
          </span>
          
        •
        
          <span
            aria-label="100, 200, 300, regular, 500, 600, 700"
          >
            7 variants
          </span>
          
        •
        
          <span
            aria-label="latin"
          >
            1 subsets
          </span>
          
      
        </div>
        
      
      </div>
      
    
    </div>
    
    
    <div
      class="family-tags"
    >
      
      
      <div
        class="family-tags-container"
      />
      
      
      <div
        class="family-meta-links"
      >
        
        
        <a
          aria-label="Edit tags for Material Symbols Sharp"
          href="https://github.com/katydecorah/font-library/blob/gh-pages/families.json#L752"
          target="_blank"
        >
          Edit tags
        </a>
        
        
        <a
          aria-label="Visit Material Symbols Sharp on Google Fonts"
          href="https://fonts.google.com/specimen/Material+Symbols+Sharp"
          target="_blank"
        >
          Google Fonts →
        </a>
        
      
      </div>
      
    
    </div>
    
  
  </div>
</font-item>
`;

exports[`FontItem is variable font 1`] = `
<font-item
  font="{"family":"Akshar","variants":["300","regular","500","600","700"],"subsets":["devanagari","latin","latin-ext"],"category":"sans-serif","tags":["condensed","narrow"],"lineNumber":17,"variable":true}"
  id="akshar"
  selectedsubset=""
  selectedvariant=""
>
  <div
    class="family"
    id="family-akshar"
  >
    
    
    <div
      class="family-link"
    >
      
      
      <div
        class="family-title akshar"
        id="family-name"
        style="font-family: 'Akshar';"
      >
        
        Akshar
      
      </div>
      
      
      <div
        class="family-meta-container"
      >
        
     
        <span
          class="family-title-small"
        />
        
      
        <div
          class="family-meta"
        >
          
        
          <span>
            sans-serif
          </span>
          
        •
        
          <span
            aria-label="300, regular, 500, 600, 700"
          >
            5 variants (variable)
          </span>
          
        •
        
          <span
            aria-label="devanagari, latin, latin-ext"
          >
            3 subsets
          </span>
          
      
        </div>
        
      
      </div>
      
    
    </div>
    
    
    <div
      class="family-tags"
    >
      
      
      <div
        class="family-tags-container"
      >
        <button
          is="tag-button"
          selectedtag="null"
          value="condensed"
        >
          condensed
        </button>
        <button
          is="tag-button"
          selectedtag="null"
          value="narrow"
        >
          narrow
        </button>
      </div>
      
      
      <div
        class="family-meta-links"
      >
        
        
        <a
          aria-label="Edit tags for Akshar"
          href="https://github.com/katydecorah/font-library/blob/gh-pages/families.json#L17"
          target="_blank"
        >
          Edit tags
        </a>
        
        
        <a
          aria-label="Visit Akshar on Google Fonts"
          href="https://fonts.google.com/specimen/Akshar"
          target="_blank"
        >
          Google Fonts →
        </a>
        
      
      </div>
      
    
    </div>
    
  
  </div>
</font-item>
`;

exports[`FontItem no latin subset 1`] = `
<font-item
  font="{"family":"GFS Neohellenic","variants":["regular","italic","700","700italic"],"subsets":["greek"],"category":"sans-serif","tags":["round","monolinear","1930s"],"lineNumber":439}"
  id="gfs-neohellenic"
  selectedsubset=""
  selectedvariant=""
>
  <div
    class="family"
    id="family-gfs-neohellenic"
  >
    
    
    <div
      class="family-link"
    >
      
      
      <div
        class="family-title gfs-neohellenic"
        id="family-name"
        style="font-family: 'GFS Neohellenic';"
      >
        
        Γειά σου Κόσμε
      
      </div>
      
      
      <div
        class="family-meta-container"
      >
        
     
        <span
          class="family-title-small"
        >
          GFS Neohellenic
        </span>
        
      
        <div
          class="family-meta"
        >
          
        
          <span>
            sans-serif
          </span>
          
        •
        
          <span
            aria-label="regular, italic, 700, 700italic"
          >
            4 variants
          </span>
          
        •
        
          <span
            aria-label="greek"
          >
            1 subsets
          </span>
          
      
        </div>
        
      
      </div>
      
    
    </div>
    
    
    <div
      class="family-tags"
    >
      
      
      <div
        class="family-tags-container"
      >
        <button
          is="tag-button"
          selectedtag="null"
          value="round"
        >
          round
        </button>
        <button
          is="tag-button"
          selectedtag="null"
          value="monolinear"
        >
          monolinear
        </button>
        <button
          is="tag-button"
          selectedtag="null"
          value="1930s"
        >
          1930s
        </button>
      </div>
      
      
      <div
        class="family-meta-links"
      >
        
        
        <a
          aria-label="Edit tags for GFS Neohellenic"
          href="https://github.com/katydecorah/font-library/blob/gh-pages/families.json#L439"
          target="_blank"
        >
          Edit tags
        </a>
        
        
        <a
          aria-label="Visit GFS Neohellenic on Google Fonts"
          href="https://fonts.google.com/specimen/GFS+Neohellenic"
          target="_blank"
        >
          Google Fonts →
        </a>
        
      
      </div>
      
    
    </div>
    
  
  </div>
</font-item>
`;

exports[`FontItem renders correctly 1`] = `
<font-item
  font="{"family":"Alef","variants":["regular","700"],"subsets":["hebrew","latin"],"category":"sans-serif","tags":["geometric"],"lineNumber":23}"
  id="alef"
  selectedsubset=""
  selectedvariant=""
>
  <div
    class="family"
    id="family-alef"
  >
    
    
    <div
      class="family-link"
    >
      
      
      <div
        class="family-title alef"
        id="family-name"
        style="font-family: 'Alef';"
      >
        
        Alef
      
      </div>
      
      
      <div
        class="family-meta-container"
      >
        
     
        <span
          class="family-title-small"
        />
        
      
        <div
          class="family-meta"
        >
          
        
          <span>
            sans-serif
          </span>
          
        •
        
          <span
            aria-label="regular, 700"
          >
            2 variants
          </span>
          
        •
        
          <span
            aria-label="hebrew, latin"
          >
            2 subsets
          </span>
          
      
        </div>
        
      
      </div>
      
    
    </div>
    
    
    <div
      class="family-tags"
    >
      
      
      <div
        class="family-tags-container"
      >
        <button
          is="tag-button"
          selectedtag="null"
          value="geometric"
        >
          geometric
        </button>
      </div>
      
      
      <div
        class="family-meta-links"
      >
        
        
        <a
          aria-label="Edit tags for Alef"
          href="https://github.com/katydecorah/font-library/blob/gh-pages/families.json#L23"
          target="_blank"
        >
          Edit tags
        </a>
        
        
        <a
          aria-label="Visit Alef on Google Fonts"
          href="https://fonts.google.com/specimen/Alef"
          target="_blank"
        >
          Google Fonts →
        </a>
        
      
      </div>
      
    
    </div>
    
  
  </div>
</font-item>
`;

exports[`FontItem rtl subset 1`] = `
<font-item
  font="{"family":"Almarai","variants":["300","regular","700","800"],"subsets":["arabic"],"category":"sans-serif","tags":["arabic"],"lineNumber":41}"
  id="almarai"
  selectedsubset=""
  selectedvariant=""
>
  <div
    class="family"
    id="family-almarai"
  >
    
    
    <div
      class="family-link"
    >
      
      
      <div
        class="family-title almarai"
        id="family-name"
        style="font-family: 'Almarai';direction: rtl;"
      >
        
        مرحبا بالعالم
      
      </div>
      
      
      <div
        class="family-meta-container"
      >
        
     
        <span
          class="family-title-small"
        >
          Almarai
        </span>
        
      
        <div
          class="family-meta"
        >
          
        
          <span>
            sans-serif
          </span>
          
        •
        
          <span
            aria-label="300, regular, 700, 800"
          >
            4 variants
          </span>
          
        •
        
          <span
            aria-label="arabic"
          >
            1 subsets
          </span>
          
      
        </div>
        
      
      </div>
      
    
    </div>
    
    
    <div
      class="family-tags"
    >
      
      
      <div
        class="family-tags-container"
      >
        <button
          is="tag-button"
          selectedtag="null"
          value="arabic"
        >
          arabic
        </button>
      </div>
      
      
      <div
        class="family-meta-links"
      >
        
        
        <a
          aria-label="Edit tags for Almarai"
          href="https://github.com/katydecorah/font-library/blob/gh-pages/families.json#L41"
          target="_blank"
        >
          Edit tags
        </a>
        
        
        <a
          aria-label="Visit Almarai on Google Fonts"
          href="https://fonts.google.com/specimen/Almarai"
          target="_blank"
        >
          Google Fonts →
        </a>
        
      
      </div>
      
    
    </div>
    
  
  </div>
</font-item>
`;

exports[`FontItem selected subset 1`] = `
<font-item
  font="{"family":"Alef","variants":["regular","700"],"subsets":["hebrew","latin"],"category":"sans-serif","tags":["geometric"],"lineNumber":23}"
  id="alef"
  selectedsubset="hebrew"
  selectedvariant=""
>
  <div
    class="family"
    id="family-alef"
  >
    
    
    <div
      class="family-link"
    >
      
      
      <div
        class="family-title alef"
        id="family-name"
        style="font-family: 'Alef';direction: rtl;"
      >
        
        שָׁלוֹם עוֹלָם 
      
      </div>
      
      
      <div
        class="family-meta-container"
      >
        
     
        <span
          class="family-title-small"
        >
          Alef
        </span>
        
      
        <div
          class="family-meta"
        >
          
        
          <span>
            sans-serif
          </span>
          
        •
        
          <span
            aria-label="regular, 700"
          >
            2 variants
          </span>
          
        •
        
          <span
            aria-label="hebrew, latin"
          >
            2 subsets
          </span>
          
      
        </div>
        
      
      </div>
      
    
    </div>
    
    
    <div
      class="family-tags"
    >
      
      
      <div
        class="family-tags-container"
      >
        <button
          is="tag-button"
          selectedtag="null"
          value="geometric"
        >
          geometric
        </button>
      </div>
      
      
      <div
        class="family-meta-links"
      >
        
        
        <a
          aria-label="Edit tags for Alef"
          href="https://github.com/katydecorah/font-library/blob/gh-pages/families.json#L23"
          target="_blank"
        >
          Edit tags
        </a>
        
        
        <a
          aria-label="Visit Alef on Google Fonts"
          href="https://fonts.google.com/specimen/Alef"
          target="_blank"
        >
          Google Fonts →
        </a>
        
      
      </div>
      
    
    </div>
    
  
  </div>
</font-item>
`;

exports[`FontItem selected variant, 900 1`] = `
<font-item
  font="{"family":"Anybody","variants":["100","200","300","regular","500","600","700","800","900","100italic","200italic","300italic","italic","500italic","600italic","700italic","800italic","900italic"],"subsets":["latin","latin-ext","vietnamese"],"category":"display","tags":["eurostile"],"lineNumber":78}"
  id="anybody"
  selectedsubset=""
  selectedvariant="900"
>
  <div
    class="family"
    id="family-anybody"
  >
    
    
    <div
      class="family-link"
    >
      
      
      <div
        class="family-title anybody"
        id="family-name"
        style="font-family: 'Anybody';"
      >
        
        Anybody
      
      </div>
      
      
      <div
        class="family-meta-container"
      >
        
     
        <span
          class="family-title-small"
        />
        
      
        <div
          class="family-meta"
        >
          
        
          <span>
            display
          </span>
          
        •
        
          <span
            aria-label="100, 200, 300, regular, 500, 600, 700, 800, 900, 100italic, 200italic, 300italic, italic, 500italic, 600italic, 700italic, 800italic, 900italic"
          >
            18 variants
          </span>
          
        •
        
          <span
            aria-label="latin, latin-ext, vietnamese"
          >
            3 subsets
          </span>
          
      
        </div>
        
      
      </div>
      
    
    </div>
    
    
    <div
      class="family-tags"
    >
      
      
      <div
        class="family-tags-container"
      >
        <button
          is="tag-button"
          selectedtag="null"
          value="eurostile"
        >
          eurostile
        </button>
      </div>
      
      
      <div
        class="family-meta-links"
      >
        
        
        <a
          aria-label="Edit tags for Anybody"
          href="https://github.com/katydecorah/font-library/blob/gh-pages/families.json#L78"
          target="_blank"
        >
          Edit tags
        </a>
        
        
        <a
          aria-label="Visit Anybody on Google Fonts"
          href="https://fonts.google.com/specimen/Anybody"
          target="_blank"
        >
          Google Fonts →
        </a>
        
      
      </div>
      
    
    </div>
    
  
  </div>
</font-item>
`;

exports[`FontItem selected variant, 900italic 1`] = `
<font-item
  font="{"family":"Anybody","variants":["100","200","300","regular","500","600","700","800","900","100italic","200italic","300italic","italic","500italic","600italic","700italic","800italic","900italic"],"subsets":["latin","latin-ext","vietnamese"],"category":"display","tags":["eurostile"],"lineNumber":78}"
  id="anybody"
  selectedsubset=""
  selectedvariant="900italic"
>
  <div
    class="family"
    id="family-anybody"
  >
    
    
    <div
      class="family-link"
    >
      
      
      <div
        class="family-title anybody"
        id="family-name"
        style="font-family: 'Anybody';font-style: italic;"
      >
        
        Anybody
      
      </div>
      
      
      <div
        class="family-meta-container"
      >
        
     
        <span
          class="family-title-small"
        />
        
      
        <div
          class="family-meta"
        >
          
        
          <span>
            display
          </span>
          
        •
        
          <span
            aria-label="100, 200, 300, regular, 500, 600, 700, 800, 900, 100italic, 200italic, 300italic, italic, 500italic, 600italic, 700italic, 800italic, 900italic"
          >
            18 variants
          </span>
          
        •
        
          <span
            aria-label="latin, latin-ext, vietnamese"
          >
            3 subsets
          </span>
          
      
        </div>
        
      
      </div>
      
    
    </div>
    
    
    <div
      class="family-tags"
    >
      
      
      <div
        class="family-tags-container"
      >
        <button
          is="tag-button"
          selectedtag="null"
          value="eurostile"
        >
          eurostile
        </button>
      </div>
      
      
      <div
        class="family-meta-links"
      >
        
        
        <a
          aria-label="Edit tags for Anybody"
          href="https://github.com/katydecorah/font-library/blob/gh-pages/families.json#L78"
          target="_blank"
        >
          Edit tags
        </a>
        
        
        <a
          aria-label="Visit Anybody on Google Fonts"
          href="https://fonts.google.com/specimen/Anybody"
          target="_blank"
        >
          Google Fonts →
        </a>
        
      
      </div>
      
    
    </div>
    
  
  </div>
</font-item>
`;
